lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -639,6 +639,24 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
            isa<ConstructorDecl>(DeclCtx);
   }
 
+  void createImportedModule(llvm::DIScope *Context,
+                            ModuleDecl::ImportedModule M, llvm::DIFile *File,
+                            unsigned Line) {
+    // For cross-import overlay modules also emit an import of the underlying
```

Comment:
micro-nit: This seems to apply to regular overlays, not cross-import overlays, right?

Reply:
Maybe I'm getting confused about the terminology here: I thought that cross-import overlay is synonymous with that it has an underlying Clang module?

Reply:
Nope, that's just a regular overlay. A cross-import overlay is e.g. the `_MapKit_SwiftUI` module in the SDK, it doesn't necessarily need to involve clang modules at all.

Reply:
Yeah, but the code explicitly checks for an underlying Clang module so I'm not yet sure I understand the concern?

Reply:
Oh the only concern is that the comment is misleading 😅

Reply:
That's what I don't get (yet) — the comment says cross-module overlays get an extra reference to the underlying clang module and the code only applies to modules with underlying clang modules.

Reply:
Right, but does this _only_ apply to cross-import overlays of clang modules? This seems to apply mostly to _regular_ overlays, I can't see anything that makes this specific to cross-import overlays.

Reply:
Can you explain what the difference between a cross-import overlay of a clang module and a regular overlay of a clang module is? This is what isn't clear to me.

Reply:
A cross-import of a clang module would involve additional metadata bundled with the swiftmodule structure that describes the other modules it can cross-over with. I don't know if we have any examples of one half of the cross being a pure clang module that is not adulterated by a Swift overlay module.

A plain overlay is a Swift module that imports an underlying clang module with the same name and extends declarations contained therein.

Reply:
Ok! So "Cross" is an orthogonal concept to the source language of the module. Which is what Harlan said a couple of comment above I now realize... thanks!